project-icp/bee-pollinator-app
# Initialize Redux with Persist

## Overview

Adds basic Redux setup using the [sample state](https://github.com/project-icp/bee-pollinator-app/issues/292#issuecomment-431479309) and Redux Persist. The setup for Redux Persist was taken mostly from [UWR](https://github.com/azavea/pwd-urban-water-revealed/blob/1.0.0/src/app/js/src/store.js), with some changes recommended in the [v5 Migration Guide](https://github.com/rt2zz/redux-persist/blob/208c6b112ead87b3701dfacaae2cdbe78377775a/docs/MigrationGuide-v5.md), particularly around `autoRehydrate`.

Connects #296 

### Demo

![image](https://user-images.githubusercontent.com/1430060/47591506-dd125a00-d93d-11e8-91f8-9dedc168a626.png)

## Testing Instructions

* Check out this branch
* Run `./scripts/beekeepers.sh install` to update yarn libraries
* Run `./scripts/beekeepers.sh start` to run the app and go to [:8000/?beekeepers](http://localhost:8000/?beekeepers) to see it
* Inspect the console. Ensure you see `persist/` actions.
* Inspect the localStorage. Ensure you see the initial state saved there.
